src/utils/config_filters.hpp

```diff
@@ -69,4 +71,10 @@ bool string_matches_if_present(
 
 bool bool_or_empty(const config& filter, const config& cfg, const std::string& attribute);
 
+bool matches_ability_filter(const config & cfg, const std::string& tag_name, const config & filter);
+
+bool common_matches_filter(const config & cfg, const std::string& tag_name, const config & filter);
+
+const std::set<std::string> exclude_tags_{"hides", "teleport", "plague", "slow", "petrifies", "firststrike", "poison", "damage_type"};
```

Comment:
What does exclude_tags_ mean? This needs a meaningful name.

Reply:
changed to 'type_excluded'

Reply:
That really doesn't say much either. Why are these tag names in this set? Presumably they have something in common?

Reply:
the numerical attribute can't be used in these abilities, but 'dummy' abilities can have these attribute, this why the term'exlude' is used forname this set

Reply:
exlude_tags removed